### ISSUE TITLE
On Linux, CUDA device names in the launch GUI are backwards from what they should be

### ISSUE THREAD

**Issue by valadaptive** [LostRuins/koboldcpp#1023]
**Describe the Issue**

I have two GPUs: a GTX 1660 Super which is connected to my monitor, and an RTX 3090 which I use for CUDA tasks.

In the KoboldCpp launcher, the first GPU (ID 1 in the launcher) is the 1660 Super, and the second GPU (ID 2) is the 3090:

![image](https://github.com/user-attachments/assets/a0000917-73c8-452f-b6c3-cacb5710b9b7)
![image](https://github.com/user-attachments/assets/25705fe4-7b28-4f6e-b003-4524710bf24d)

This matches with the output of `nvidia-smi`, which is how the launcher determines GPU indices:
```bash
$ nvidia-smi --query-gpu=name,memory.total,index --format csv,noheader
NVIDIA GeForce GTX 1660 SUPER, 6144 MiB, 0
NVIDIA GeForce RTX 3090, 24576 MiB, 1
```

However, in llama.cpp, the order is reversed:
```
ggml_cuda_init: found 2 CUDA devices:
  Device 0: NVIDIA GeForce RTX 3090, compute capability 8.6, VMM: yes
  Device 1: NVIDIA GeForce GTX 1660 SUPER, compute capability 7.5, VMM: yes
```

This means that if I select the ID corresponding to the 1660 Super in the launcher, it will actually use the 3090, and vice versa.

**Additional Information:**

Fedora 40; koboldcpp 1.71 and current `concedo` branch both exhibit this bug

**Comment by JimikoSK** [comment]
I noticed this too, my Quadro P600 should be first, but it was last.

```
ggml_cuda_init: found 3 CUDA devices:
  Device 0: Tesla P40, compute capability 6.1, VMM: yes
  Device 1: Tesla P40, compute capability 6.1, VMM: yes
  Device 2: Quadro P600, compute capability 6.1, VMM: yes
```

**Comment by LostRuins** [comment]
Hmm this is a known issue

https://stackoverflow.com/questions/52815708/order-of-cuda-devices

CUDA provides devices by order of speed by default, while nvidia-smi orders them by PCI index.

**Comment by LostRuins** [comment]
I can force them to match the same order with ` os.environ["CUDA_DEVICE_ORDER"]="PCI_BUS_ID"`, but I'm hesitant to do so as it would mean the GPU ids in kcpp and lcpp will no longer match. It would also mean that the default selected GPU is not going to be the best one.

@valadaptive can you share your launch flags? When running with a single GPU selected, I set `os.environ["CUDA_VISIBLE_DEVICES"] = "0"` to force only 1 GPU to be visible - so that should work regardless of what GPU is the fastest, it should pick by PCI ID.

**Comment by valadaptive** [comment]
I'm not passing any command-line flags when I run the koboldcpp executable. The flags that end up getting passed from the launcher are:

`Namespace(benchmark=None, blasbatchsize=512, blasthreads=7, chatcompletionsadapter=None, config=None, contextsize=4096, debugmode=0, flashattention=False, forceversion=0, foreground=False, gpulayers=43, highpriority=False, hordeconfig=None, hordegenlen=0, hordekey='', hordemaxctx=0, hordemodelname='', hordeworkername='', host='', ignoremissing=False, launch=True, lora=None, mmproj=None, model='', model_param='/media/Data/kcpp-models/codellama-13b-instruct.Q4_K_M.gguf', multiuser=1, noavx2=False, noblas=False, nocertify=False, nommap=False, noshift=False, onready='', password=None, port=5001, port_param=5001, preloadstory=None, quantkv=0, quiet=False, remotetunnel=False, ropeconfig=[0.0, 10000.0], sdclamped=0, sdconfig=None, sdlora='', sdloramult=1.0, sdmodel='', sdquant=False, sdthreads=7, sdvae='', sdvaeauto=False, skiplauncher=False, smartcontext=False, ssl=None, tensor_split=None, threads=7, unpack='', useclblast=None, usecublas=['normal', '0', 'mmq'], usemlock=False, usevulkan=None, whispermodel='')`

llama.cpp does only "see" one CUDA device when I select a specific one--the problem is that it sees the wrong one. For example, when I select the 1660 in the GUI, llama.cpp logs:
```
ggml_cuda_init: found 1 CUDA devices:
  Device 0: NVIDIA GeForce RTX 3090, compute capability 8.6, VMM: yes
```

Is it possible that `CUDA_VISIBLE_DEVICES` also uses CUDA's ordering?

**Comment by LostRuins** [comment]
Yeah that seems like a possibility. Looks like I do need to set `os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"`. Let me try that.

**Comment by LostRuins** [comment]
Hi, please try the test binary here: https://github.com/LostRuins/koboldcpp/actions/runs/10164784312 (you may need to login to github to download it)

**Comment by valadaptive** [comment]
Hey, sorry for the delay. I can't run the test binary because I'm on Linux, but I built it myself and can confirm that `os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"` fixes the issue.

If you want to make the GPUs in kcpp and lcpp match, it seems like you can get the list of CUDA devices from llama.cpp (in CUDA's order) using `ggml_backend_cuda_get_device_count` and `ggml_backend_cuda_get_device_description`.

**Comment by LostRuins** [comment]
Hi, this should be fixed in 1.72, please try!
